### PR TITLE
drivers: i2s frame clock slave mode

### PR DESCRIPTION
**Pull request by slawblauciak**
Signed-off-by: Slawomir Blauciak <slawomir.blauciak@linux.intel.com>

### CODE REVIEW

File: src/drivers/apl-ssp.c [hunk 1]
```diff
@@ -215,11 +215,7 @@ static inline int ssp_set_config(struct dai *dai,
 	switch (config->format & SOF_DAI_FMT_MASTER_MASK) {
 	case SOF_DAI_FMT_CBM_CFM:
 		sscr0 |= SSCR0_ECS; /* external clock used */
-		sscr1 |= SSCR1_SCLKDIR;
-		/*
-		 * FIXME: does SSRC1.SCFR need to be set
-		 * when codec is master ?
-		 */
+		sscr1 |= SSCR1_SCLKDIR | SSCR1_SFRMDIR;
```
Comment (plbossart) [inline src/drivers/apl-ssp.c]: yes, but then remove line 211/212 as well while you are at it.

File: src/drivers/apl-ssp.c [hunk 2]
```diff
@@ -228,10 +224,6 @@ static inline int ssp_set_config(struct dai *dai,
 	case SOF_DAI_FMT_CBM_CFS:
 		sscr0 |= SSCR0_ECS; /* external clock used */
```
Comment (plbossart) [inline src/drivers/apl-ssp.c]: remove this, this has nothing to do with master/slave mode.

File: src/drivers/apl-ssp.c [hunk 3]
```diff
@@ -215,11 +215,7 @@ static inline int ssp_set_config(struct dai *dai,
...
-		sscr1 |= SSCR1_SCLKDIR;
-		/*
-		 * FIXME: does SSRC1.SCFR need to be set
-		 * when codec is master ?
-		 */
+		sscr1 |= SSCR1_SCLKDIR | SSCR1_SFRMDIR;
 		break;
 	case SOF_DAI_FMT_CBS_CFS:
 		sscr1 |= SSCR1_SCFR;
```
Comment (plbossart) [inline src/drivers/apl-ssp.c]: remove this, this has nothing to with with SOF master mode.

**Comment by slawblauciak** [comment]
Changes applied as suggested

**Comment by slawblauciak** [comment]
Yeah, all good

**Comment by plbossart** [comment]
ok, let's merge this. I'll check on LeafHill later.

**Pull request merged**